packages/components/src/__actions__/Button/v3/_docs/Button.spec.stories.tsx

```diff
@@ -1,131 +1,126 @@
-
-      await waitFor(() => expect(document.activeElement).toBe(button))
-      await expect(button).toHaveAttribute("data-focused", "true")
-    })
+    expect(button).toHaveAccessibleName("Hidden label is accessible")
   },
 }
 
-export const IsFocusedReversed: Story = {
-  ...IsFocused,
-  parameters: {
-    reverseColors: true,
+export const PendingButton: Story = {
+  render: ({ isPending = false, pendingLabel = "Loading", ...otherProps }) => {
+    const [isPendingStatus, setIsPendingStatus] =
+      React.useState<boolean>(isPending)
+
+    return (
+      <Button
+        {...otherProps}
+        isPending={isPendingStatus}
+        pendingLabel={pendingLabel}
+        onPress={() => {
+          setIsPendingStatus(true)
+          setTimeout(() => {
+            setIsPendingStatus(false)
+          }, 900)
+        }}
+      />
+    )
   },
-}
-
-export const IsPressed: Story = {
-  ...ButtonDefault,
   play: async ({ canvasElement, step }) => {
     const canvas = within(canvasElement.parentElement!)
-    const button = canvas.getByRole("button")
+    const button = canvas.getByRole("button", { name: "Label" })
+    // Simulates a delay that may occur when a button must wait for a response
+    const timeToWait = 1000
 
-    await step("onPress is called", async () => {
+    await step("Button has accessible label", async () =>
+      expect(button).toHaveAccessibleName("Label")
+    )
+
+    await step("Accessible label updates on press", async () => {
+      await button.focus()
       await userEvent.click(button)
+      await waitFor(() => expect(button).toHaveAccessibleName("Loading"))
+    })
 
-      await expect(onPressEvent).toHaveBeenCalledTimes(1)
+    await step("Accessible label reverts once isPending is false", async () => {
+      await waitFor(() =>
+        setTimeout(() => {
+          expect(button).toHaveAccessibleName("Label")
+        }, timeToWait)
+      )
     })
   },
 }
 
-export const IconButtonWithAccessibleLabel: Story = {
-  render: ({ children: _, ...otherArgs }) => (
-    <Button {...otherArgs}>
-      <>
-        <TrashIcon role="img" aria-label="Remove" />
-        <VisuallyHidden> Highlight: 18, June, 2024</VisuallyHidden>
-      </>
-    </Button>
-  ),
-  play: async ({ canvasElement, step }) => {
-    const canvas = within(canvasElement.parentElement!)
-    const button = canvas.getByRole("button")
-
-    await step("has accessible labels", async () => {
-      await userEvent.tab()
+export const PendingButtonWithHiddenPendingLabel: Story = {
+  ...PendingButton,
+  args: {
+    hasHiddenPendingLabel: true,
+    isPending: false,
+    pendingLabel: "Loading",
+  },
+}
 
-      await expect(button).toHaveAccessibleName(
-        "Remove Highlight: 18, June, 2024"
-      )
-    })
+export const PendingIconButton: Story = {
+  ...PendingButton,
+  args: {
+    hasHiddenLabel: true,
+    isPending: false,
+    pendingLabel: "Loading",
+    icon: <Icon name="add" isPresentational />,
   },
 }
```

Comment:
You should be able to move these into the stickersheet

Reply:
Most have been covered but will add the scenario for maintaining the button's content width